src/app/ReadHandler.cpp

```diff
@@ -753,42 +754,57 @@ void ReadHandler::PersistSubscription()
     }
 }
 
-void ReadHandler::OnUnblockHoldReportCallback(System::Layer * apSystemLayer, void * apAppState)
+void ReadHandler::OnReportOnMinCallback(System::Layer * apSystemLayer, void * apAppState)
 {
     VerifyOrReturn(apAppState != nullptr);
     ReadHandler * readHandler = static_cast<ReadHandler *>(apAppState);
-    ChipLogDetail(DataManagement, "Unblock report hold after min %d seconds", readHandler->mMinIntervalFloorSeconds);
+    // This should wake after the minimal interval is expired, making this flag redundant. this is only kept to accomodate the unit
+    // test
     readHandler->ClearStateFlag(ReadHandlerFlags::HoldReport);
-    InteractionModelEngine::GetInstance()->GetExchangeManager()->GetSessionManager()->SystemLayer()->StartTimer(
-        System::Clock::Seconds16(readHandler->mMaxInterval - readHandler->mMinIntervalFloorSeconds),
-        OnRefreshSubscribeTimerSyncCallback, readHandler);
+    if (readHandler->IsReportable())
+    {
+        InteractionModelEngine::GetInstance()->GetReportingEngine().ScheduleRun();
+    }
+    ChipLogProgress(DataManagement, "Refresh subscribe timer after %d seconds", readHandler->mMinIntervalFloorSeconds);
 }
 
-void ReadHandler::OnRefreshSubscribeTimerSyncCallback(System::Layer * apSystemLayer, void * apAppState)
+void ReadHandler::OnRefreshSubscribeTimerCallback(System::Layer * apSystemLayer, void * apAppState)
 {
     VerifyOrReturn(apAppState != nullptr);
     ReadHandler * readHandler = static_cast<ReadHandler *>(apAppState);
     readHandler->ClearStateFlag(ReadHandlerFlags::HoldSync);
-    ChipLogProgress(DataManagement, "Refresh subscribe timer sync after %d seconds",
-                    readHandler->mMaxInterval - readHandler->mMinIntervalFloorSeconds);
+    ChipLogProgress(DataManagement, "Refresh subscribe timer after %d seconds", readHandler->mMaxInterval);
 }
 
-CHIP_ERROR ReadHandler::RefreshSubscribeSyncTimer()
+CHIP_ERROR ReadHandler::RescheduleSubscribeTimer()
 {
     InteractionModelEngine::GetInstance()->GetExchangeManager()->GetSessionManager()->SystemLayer()->CancelTimer(
-        OnUnblockHoldReportCallback, this);
+        OnRefreshSubscribeTimerCallback, this);
+
+    ChipLogProgress(DataManagement, "Reschedule Subscribe Timer with %" PRIu64 " miliseconds",
+                    mNextMinWakeUpTime.count() - System::SystemClock().GetMonotonicTimestamp().count());
```

Comment:
- Please introduce a variable here, reused in the log and the timer, to reduce number of GetMonotonicTimestamp cost
- Please log AFTER starting the timer, so that any log delays do not jitter the timer itself.
- Also, how often is this logging? This can get quite chatty.